rust/arrow/src/csv/reader.rs

```diff
@@ -67,6 +67,9 @@ lazy_static! {
         .case_insensitive(true)
         .build()
         .unwrap();
+    static ref DATE_RE: Regex = Regex::new(r"^\d\d\d\d-\d\d-\d\d$").unwrap();
```

Comment:
isn't there a `\d{4}` or something like that? May make it a bit easier to read and more expressive, IMO